container-host-files/opt/scf/bin/klog.sh

```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] InnoDB: Cannot open table mysql/gtid_slave_pos from the internal data dictionary of InnoDB
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: no nodes coming from prim view, prim not possible
-
-Nodes all shut down improperly, MySQL must be manually rebootstrapped. Will fail
-if the IST patch has not been applied.
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: Failed to prepare for incremental state transfer
-
-IST addresses are misconfigured, likely a <= 4.0.1 cluster, manual
-patch must be applied and MySQL must be recovered manually.
-
-Not sure exactly when/why this occurs. Perhaps when the primary node
-has gone away (as was the case here). The only restitution seems to be
-to full recover the mysql node (and ensure the IST patch is done):
-
-monit stop all # Check for lingering processes and kill them
-rm -rf /var/vcap/store/mysql/
-/var/vcap/jobs/mysql/bin/pre-start
-monit start all
-EOF
-
-read -d '' MSG <<EOF || true
-These point to an etcd that came up in a weird way and thinks there
-are other nodes when there are none. Restoration was done by:
-monit stop all
-rm -rf /var/vcap/store/etcd
-# Restart the pod using kubectl
-EOF
-
-# etcd pod /var/vcap/sys/log/etcd/etcd_ctl.err.log
-lookfor <<EOF
-Error:  cannot sync with the cluster using endpoints https://etcd-
-
-${MSG}
-EOF
+if [ ! -f "${DONE}" ]; then
+    mkdir -p "${KLOG}/${NS}"
+    rm -rf "${KLOG:?}/${NS:?}/"*
 
-# etcd pod /var/vcap/sys/log/etcd/etcd.stderr.log
-# duplicate of previous error?
-lookfor <<EOF
-etcdserver: publish error: etcdserver: request timed out
+    PODS=$(kubectl get pods --namespace "${NS}" --output name --show-all=true | sed 's/pods\///')
 
-${MSG}
-EOF
+    for POD in ${PODS}; do
+        DIR=${KLOG}/${NS}/${POD}
 
-# mysql pods /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-SST disabled due to danger of data loss. Verify data and bootstrap the cluster
+        mkdir -p "${DIR}"
 
-IST is probably disabled so it's falling back to this, this happens on
-4.0.1 clusters, the fix is complicated as it involves manually
-patching.
-EOF
-
-# consul pod /var/vcap/sys/log/consul_agent/consul_agent.stdout.log
-lookfor <<EOF
-Only one node should be in bootstrap mode, not adding Raft peer.
+        # Get the CF logs inside the pod if there are any
+        if [ "$(kubectl get pod "${POD}" --namespace "${NS}" --output=jsonpath='{.status.phase}')" != 'Succeeded' ] && \
+                kubectl exec "${POD}" --namespace "${NS}" -- bash -c "[ -d /var/vcap/sys/log ]" 2> /dev/null; then
+            # Mask the exit status of tar because it complains if files were written while it was reading them
+            kubectl exec "${POD}" --namespace "${NS}" -- bash -c "cd /var/vcap/sys/log && (tar --warning=no-file-changed -cf - * || true)" | ( cd "${DIR}" && tar xf -)
```

Comment:
`-- tar -C /var/vcap/sys/log --warning=no-file-changed …`
`… tar -C "${DIR}" xf -`

But then, this is basically `kubectl cp --namespace "${NS}" "${POD}:/var/vcap/sys/log/" "${DIR}/"`

Reply:
TIL about `cp` :grinning: 

Fixed in 4442684c